### PR TITLE
Promise rejection handler

### PR DESCRIPTION
**Pull request by corps**
This is the rollbar.js side of https://github.com/rollbar/node_rollbar/issues/61

Tests are not complete yet, and documentation is not complete, this tentative pull puts out my approach.  I'm especially looking for feedback due to the complexity of browser implementation.

### CODE REVIEW

**Comment by corps** [comment]
Oops, gotta rebase this branch back off master, too.  Anyways, check the last commit for the isolated changes.

File: src/globalnotifier.js [hunk 1]
```diff
@@ -78,6 +77,17 @@ wrapper.init = function(config, parent) {
     }
   }
 
+  if (config.captureUnhandledRejections) {
+    if (parent && Util.isType(parent._unhandledRejectionHandler, 'function')) {
+      window.removeEventListener('unhandledrejection', client._unhandledRejectionHandler)
```
Comment (coryvirok) [inline src/globalnotifier.js]: Should this be `parent._unhandledRejectionHandler`?

File: src/notifier.js [hunk 2]
```diff
@@ -720,6 +720,50 @@ NotifierPrototype.uncaughtError = _wrapNotifierFn(function(message, url, lineNo,
...
+      'unhandled rejection was null or undefined!', // message
+      null, // err
+      {}, // custom
+      null,  // callback
+      false, // isUncaught
+      false); // ignoreRateLimit
+    return;
+  }
+
+  var message = reason.message || (reason ? String(reason) : (reason || 'unhandled rejection'));
```
Comment (coryvirok) [inline src/notifier.js]: `(reason || 'unhandled rejection')` can just be `'unhandled rejection'`

File: src/notifier.js [hunk 3]
```diff
@@ -720,6 +720,50 @@ NotifierPrototype.uncaughtError = _wrapNotifierFn(function(message, url, lineNo,
...
+    return;
+  }
+
+  var location = {
+    'url': '',
+    'line': 0
+  };
+  location.func = errorParser.guessFunctionName(location.url, location.line);
+  location.context = errorParser.gatherContext(location.url, location.line);
+  var stack = {
```
Comment (coryvirok) [inline src/notifier.js]: Can we not get an actual stack trace here instead?

**Comment by corps** [comment]
Closed in order to fix commit history, with new tests and fixes. See new pull  https://github.com/rollbar/rollbar.js/pull/193

**Pull request closed without merging**